### PR TITLE
Allow to configure smtp and sendmail in gitlab.yml

### PR DESCRIPTION
**Pull request by jirutka**
Reopening #8800.

TODO:
- [ ] Make it backward compatible with existing `smtp_settings.rb` (see comments [here](https://github.com/gitlabhq/gitlabhq/pull/8800#issuecomment-99359016) and [here](https://github.com/gitlabhq/gitlabhq/pull/8800#issuecomment-99494266)).
- [ ] Create `doc/update/7.10-to-7.11.md` and add instructions to migrate email settings from `smtp_settings.rb` to `gitlab.yml`.
- [ ] Mention this change in the changelog.
- [ ] Update Omnibus package → [omnibus-gitlab#555](https://gitlab.com/gitlab-org/omnibus-gitlab/issues/555).

### CODE REVIEW

File: config/gitlab.yml.example [hunk 1]
```diff
@@ -89,6 +79,57 @@ production: &base
...
+    smtp_settings:
+      # Hostname of the SMTP server used to send emails (default: localhost).
+      address: localhost
+
+      # Port of the SMTP server used to send emails (default: 25).
+      port: 25
+
+      # If your mail server requires authentication, you need to specify the
+      # authentication type here. Possible values are:
+      #   :plain    - send the password in the clear text.
```
Comment (axilleas) [inline config/gitlab.yml.example]: The `plain` method does not send the password in cleartext. See https://github.com/rails/rails/blob/master/actionmailer/lib/action_mailer/base.rb#L383 and https://github.com/rails/rails/pull/19930

Comment (dserodio) [inline config/gitlab.yml.example]: base64 _is_ cleartext. It's not encrypted or hashed

Comment (axilleas) [inline config/gitlab.yml.example]: I see your point and I agree, but we have two methods that behave almost the same, `:plain` and `:login` and they are both base64 encoded. The way it is written now, one might think `:login` is stronger than `:plain`.

I don't know how you define cleartext, but in my mind it means that a password is read as is. However in base64 you'd have to do a decode first, although trivial.

Comment (jirutka) [inline config/gitlab.yml.example]: What’s difference between `plain` and `login`?

**Comment by jvanbaarsen** [comment]
This merge request has been closed because a request for more information has not been reacted to for more than 2 weeks. If you respond and conform to the merge request guidelines in our [contributing guidelines](https://gitlab.com/gitlab-org/gitlab-ce/blob/master/CONTRIBUTING.md#pull-requests) we will reopen this merge request.

**Pull request closed without merging**